tensorly/decomposition/_cp.py

```diff
@@ -44,9 +43,7 @@ def initialize_cp(tensor, rank, init='svd', svd='numpy_svd', random_state=None,
     rng = check_random_state(random_state)
 
     if init == 'random':
-        # factors = [tl.tensor(rng.random_sample((tensor.shape[i], rank)), **tl.context(tensor)) for i in range(tl.ndim(tensor))]
-        # kt = CPTensor((None, factors))
-        return random_cp(tl.shape(tensor), rank, normalise_factors=False, **tl.context(tensor))
+        kt = random_cp(tl.shape(tensor), rank, normalise_factors=False, **tl.context(tensor))
```

Comment:
This can't immediately return because it potentially has to be normalized later.